src/main/java/bio/terra/cli/app/utils/tables/TablePrinter.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.cli.app.utils.tables;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Simple generic printer for object fields in rows separated by newlines and delimited by tabs.
+ *
+ * @param <UF_TYPE> User-facing type to be tabularized
+ */
+@FunctionalInterface
+public interface TablePrinter<UF_TYPE> {
+
+  /** Whitespace between columns. */
+  String FIELD_DELIMITER = "  ";
+
+  /** Separation between rows. */
+  String ROW_DELIMITER = "\n";
+
+  /**
+   * Functional interface caller-supplied method. Returns an array of PrintableColumns. To
+   * instantiate a TablePrinter, supply an array of `TablePrintable`s. If an Enum type implements
+   * this array, the object may be instantiated simply by {@code TablePrinter<UFWorkspace>
+   * workspaceTablePrinter = UFWorkspaceColumns::values;}. This creates an instance of an anonymous
+   * class implementing TablePrinter.
+   */
+  ColumnDefinition<UF_TYPE>[] getColumnEnumValues();
+
+  /**
+   * Print a table from a list of row objects of type T. Do not leave space for a highlight column.
+   *
+   * @param rowObjects - list of user-facing objects to print to rows of the table.
+   * @return string representation of table suitable for printing to console
+   */
+  default String print(List<UF_TYPE> rowObjects) {
+    return print(rowObjects, null);
+  }
+
+  /**
+   * Print the list of objects to a table, with their column labels, ordering, and field values
+   * determined by the enum array given by @Code {getColumnEnumValues}.
+   *
+   * @param rowObjects - list of user-facing objects to print to rows of the table.
+   * @param isHighlighted - boolean-valued function to tell if a row should be highlighted (starred)
+   * @return table string suitable for printing to console
+   */
+  default String print(List<UF_TYPE> rowObjects, @Nullable Predicate<UF_TYPE> isHighlighted) {
```

Comment:
can we just do Out.println() instead of a string since this method is called print()?

Reply:
It's slightly more unit testable and portable the way it is, e.g. if we wanted to move the printing stuff to another library, it shouldn't know anything about how the CLI outputs things.